crates/egui/src/ui.rs

```diff
@@ -97,17 +97,19 @@ impl Ui {
         id_source: impl Hash,
     ) -> Self {
         crate::egui_assert!(!max_rect.any_nan());
-        let next_auto_id_source = Id::new(self.next_auto_id_source).with("child").value();
+        let child_id = self.id.with(id_source);
+        let child_next_auto_id_source = child_id.with(self.next_auto_id_source).value();
+
         self.next_auto_id_source = self.next_auto_id_source.wrapping_add(1);
```

Comment:
As-is, this PR does not fix the problem in my application (#2253.) I think this is because the child UI's `next_auto_id_source` is still dependent on how much stuff was rendered before the child_ui. Something like this works for me:

```suggestion
        let child_next_auto_id_source = child_id.with("auto").value();
```

Reply:
The problem is `id_source` is not always unique in this instance. A UI can have many child-ui:s, all created with `child_ui` which use the same id_source (`"child"`), so we need to seed the auto-ids differently for all these children.

Applying this diff gets me:

![Screen Shot 2022-12-02 at 10 08 12](https://user-images.githubusercontent.com/1148717/205256952-f41898bd-720e-4ce2-9e95-53539fa205c0.png)